MelonLoader/Melons/Melon.cs

```diff
@@ -20,25 +18,19 @@ public abstract class Melon<T> : MelonBase where T : Melon<T>
 
         static Melon()
         {
-            System.Runtime.CompilerServices.RuntimeHelpers.RunClassConstructor(typeof(T).TypeHandle); // to make sure the type initializer was triggered
+            System.Runtime.CompilerServices.RuntimeHelpers.RunClassConstructor(typeof(T).TypeHandle); // To make sure that the type initializer of T was triggered.
         }
 
         public sealed override string MelonTypeName => TypeName;
 
-        public sealed override bool Register()
```

Comment:
This is technically an API breaking change, should consult modders to make sure nobody is using this (though I doubt it)

Reply:
Wait, can you explain what the "API breaking change" is?

Reply:
As in, you've made the method less visible, so for any mod that was calling this from outside of a melon class (which, realistically, nobody SHOULD have been), that mod is now broken.

Reply:
Wait wait what? What did i make less visible

Reply:
You changed this method from public to protected internal

Reply:
Nononononono. First of all, its still public. What you're looking at is a NEW overridable method called `RegisterInternal`.
![image](https://user-images.githubusercontent.com/61495410/154660166-8cc8d492-6edc-41e3-bb35-c024dc7e8ccd.png)

And even if I changed the visibility of it rn, it hasn't even been in a full release yet so I'm still allowed to do that